src/core/thread/discover_scanner.cpp

```diff
@@ -144,6 +157,25 @@ otError DiscoverScanner::Discover(const Mac::ChannelMask &aScanChannels,
     return error;
 }
 
+otError DiscoverScanner::SetJoinerAdvertisement(uint32_t aOui, const uint8_t *aAdvData, uint8_t aAdvDataLength)
+{
+    otError error = OT_ERROR_NONE;
+
+    if ((aAdvData == nullptr) || (aAdvDataLength == 0) || (aAdvDataLength > kMaxLength))
+    {
+        error = OT_ERROR_INVALID_ARGS;
+        return error;
```

Comment:
nit: https://github.com/openthread/openthread/blob/master/STYLE_GUIDE.md
> Return Statements
There should be one return statement per free function or method at the end of the free function or method.